exercises/linked-list/canonical-data.json

```diff
@@ -0,0 +1,532 @@
+{
+  "exercise": "linked-list",
+  "comments": [
+    "In order to keep the interfae for the exercise close or equal to the     ",
+    "description.md and also satisfying the canonical-schema.json, the only   ",
+    "properties are pop, push, shift, unshift. Some tracks also implement     ",
+    "delete and count, via a Track-Inserted hints.                            ",
+    "                                                                         ",
+    "It is hard to have interesting tests using the property based approach so",
+    "this canonical data uses the following approach:                         ",
+    "- Each test input is an array of { operation, value?, expected? }        ",
+    "- If operation is push, unshift, delete, then value is given             ",
+    "- If operation is pop, shift, count, the expected value can be given     ",
+    "                                                                         ",
+    "Encoding tests and operations using the same field is necessary to have  ",
+    "tests that don't just initialize the list, and then have one operation.  "
+  ],
+  "cases": [
+    {
+      "uuid": "7f7e3987-b954-41b8-8084-99beca08752c",
+      "description": "pop gets element from the list",
+      "property": "list",
+      "input": {
+        "operations": [
+          {
+            "operation": "push",
+            "value": 7
+          },
+          {
+            "operation": "pop",
+            "expected": 7
+          }
+        ]
+      },
+      "expected": []
+    },
+    {
+      "uuid": "c3f67e5d-cfa2-4c3e-a18f-7ce999c3c885",
+      "description": "push/pop respectively add/remove at the end of the list",
+      "property": "list",
+      "input": {
+        "operations": [
+          {
+            "operation": "push",
+            "value": 11
+          },
+          {
+            "operation": "push",
+            "value": 13
+          },
+          {
+            "operation": "pop",
+            "expected": 13
+          },
+          {
+            "operation": "pop",
+            "expected": 11
+          }
+        ]
+      },
+      "expected": []
+    },
+    {
+      "uuid": "00ea24ce-4f5c-4432-abb4-cc6e85462657",
+      "description": "shift gets element from the list",
```

Comment:
```suggestion
      "description": "shift gets first element from the list with only one element",
```

Reply:
No, it's purposefully testing you get "a" element. The next test tests that you get the correct element.

Reply:
I should probably add this as comments?

Reply:
Might be useful. But perhaps I'm the only one being tripped up by this 🤷